modules/platforms/cpp/thin-client/include/ignite/thin/transactions/transactions.h

```diff
@@ -0,0 +1,125 @@
+             *
+             * This is an entry point for Thin C++ Ignite transactions.
+             *
+             * This class implemented as a reference to an implementation so copying of this class instance will only
+             * create another reference to the same underlying object. Underlying object released automatically once all
+             * the instances are destructed.
+             */
+            class ClientTransactions {
+            public:
+                /**
+                 * Constructor.
+                 *
+                 * @param impl Implementation.
+                 */
+                ClientTransactions(ignite::common::concurrent::SharedPointer<void> impl) :
+                    proxy(impl),
+                    label(ignite::common::concurrent::SharedPointer<ignite::common::FixedSizeArray<char> >())
+                {
+                }
+
+                /**
+                 * Destructor.
+                 */
+                ~ClientTransactions()
+                {
+                }
+
+                /**
+                 * Start new transaction with completely clarify parameters.
+                 *
+                 * @param concurrency Transaction concurrency.
+                 * @param isolation Transaction isolation.
+                 * @param timeout Transaction timeout.
+                 * @param txSize Number of entries participating in transaction (may be approximate).
+                 *
+                 * @return ClientTransaction implementation.
+                 */
+                ClientTransaction TxStart(
+                        TransactionConcurrency::Type concurrency = TransactionConcurrency::PESSIMISTIC,
+                        TransactionIsolation::Type isolation = TransactionIsolation::READ_COMMITTED,
+                        int64_t timeout = 0,
+                        int32_t txSize = 0)
+                {
+                    return ClientTransaction(proxy.txStart(concurrency, isolation, timeout, txSize, label));
+                }
+
+                /**
+                 * Returns instance of {@code ClientTransactions} to mark each new transaction with a specified label.
+                 *
+                 * @param label Transaction label.
+                 * @return ClientTransactions implementation.
+                 */
+                ClientTransactions withLabel(const std::string& lbl)
+                {
+                    ClientTransactions copy = ClientTransactions(proxy, lbl);
+
+                    return copy;
+                }
+            private:
+                /** Implementation. */
+                ignite::impl::thin::transactions::TransactionsProxy proxy;
+
+                /** Transaction specific label. */
+                ignite::common::concurrent::SharedPointer<ignite::common::FixedSizeArray<char> > label;
+
+                /**
+                 * Default constructor.
+                 */
+                ClientTransactions();
```

Comment:
Not sure why should it be private. May cause additional trouble for users, let's leave it public.